core/testcontainers/core/container.py

```diff
@@ -25,11 +26,11 @@ class DockerContainer:
         ...    delay = wait_for_logs(container, "Hello from Docker!")
     """
 
-    def __init__(self, image: str, docker_client_kw: Optional[dict] = None, **kwargs) -> None:
+    def __init__(self, image: Union[DockerImage, str], docker_client_kw: Optional[dict] = None, **kwargs) -> None:
         self.env = {}
         self.ports = {}
         self.volumes = {}
-        self.image = image
+        self.image = image.get_wrapped_image() if isinstance(image, DockerImage) else image
```

Comment:
can we simplify this logic to be:
1. if string, construct a `DockerImage`
2. get a string which can be passed to `docker run`
and try to contain the conditionals inside `DockerImage`